cmd/restic/cmd_ls.go

```diff
@@ -11,12 +13,21 @@ import (
 )
 
 var cmdLs = &cobra.Command{
-	Use:   "ls [flags] [snapshot-ID ...]",
+	Use:   "ls [flags] [snapshotID] [dir...]",
 	Short: "List files in a snapshot",
 	Long: `
-The "ls" command allows listing files and directories in a snapshot.
-
-The special snapshot-ID "latest" can be used to list files and directories of the latest snapshot in the repository.
+The "ls" command lists files and directories in a snapshot.
+
+The special snapshot ID "latest" can be used to list files and
+directories of the latest snapshot in the repository. The
+--host flag can be used in conjunction to select the latest
+snapshot originating from a certain host only.
+
+File listings can optionally be filtered by directories. Any
+positional arguments after the snapshot ID are interpreted as
+directory paths, and only files inside those directories will
```

Comment:
I think it'd be good to have a hint here that `dir` needs to be slash-separated, and optionally starting with a `/`. How do you want to handle that? The path passed to the function in `walker.Walk()` will be slash-separated and starting with a `/` (so it's an absolute path)...

Reply:
Hmm, what do you mean _optionally_ starting with a `/`? I can mention it, I just don't quite understand what we need to communicate here yet.

Reply:
Ah, sorry, I should have been clearer: the filter functions you wrote only work when the paths passed to `restic ls` are absolute (starting with a `/`), because the `nodepath` passed to the function in `walker.Walk()` starts with a slash. So I think we should document it.

Another option would be to just prefix the dirs with a slash if it isn't there already. That's what I meant with "optionally". Thinking about it, I'd suggest just requiring the paths to be absolute.